safenode/src/protocol/types/wallet/local_store.rs

```diff
@@ -146,10 +149,14 @@ impl<C: SendClient + Send + Sync + Clone> SendWallet<C> for LocalSendWallet<C> {
 
         let mut available_dbcs = vec![];
         for dbc in self.wallet.available_dbcs.values() {
-            let derived_key = dbc
-                .derived_key(&self.wallet.key)
-                .expect("This dbc to be ours.");
-            available_dbcs.push((dbc.clone(), derived_key));
+            if let Ok(derived_key) = dbc.derived_key(&self.wallet.key) {
+                available_dbcs.push((dbc.clone(), derived_key));
+            } else {
+                warn!(
+                    "Skipping DBC {:?} because we don't have the key to spend it",
+                    dbc.id()
+                );
+            }
         }
```

Comment:
Having DBCs that are not ours in our wallet should be a bug right? While having `expect` makes our code panic prone and we should indeed remove them, we should also note that this change potentially hides this bug from our eyes. 
Or is this a feature?

Reply:
The wallet we have implemented does not add dbcs that are not for us.
We don't use expect and panics in production code, not for bugs either.

So, this is unreachable code for us. And it just needs some handling to compile.. so a warn log..